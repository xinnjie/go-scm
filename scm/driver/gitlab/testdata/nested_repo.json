{
    "id": 31639063,
    "description": "",
    "name": "Gitlab Import Test 1",
    "name_with_namespace": "jx-gitlab-test / cluster / Gitlab Import Test 1",
    "path": "gitlab-import-test-1",
    "path_with_namespace": "jx-gitlab-test/cluster/gitlab-import-test-1",
    "created_at": "2021-11-26T16:41:47.275Z",
    "default_branch": "main",
    "tag_list": [],
    "topics": [],
    "ssh_url_to_repo": "git@gitlab.com:jx-gitlab-test/cluster/gitlab-import-test-1.git",
    "http_url_to_repo": "https://gitlab.com/jx-gitlab-test/cluster/gitlab-import-test-1.git",
    "web_url": "https://gitlab.com/jx-gitlab-test/cluster/gitlab-import-test-1",
    "readme_url": null,
    "avatar_url": null,
    "forks_count": 0,
    "star_count": 0,
    "last_activity_at": "2021-11-26T16:41:47.275Z",
    "namespace": {
        "id": 14253060,
        "name": "cluster",
        "path": "cluster",
        "kind": "group",
        "full_path": "jx-gitlab-test/cluster",
        "parent_id": 14253054,
        "avatar_url": null,
        "web_url": "https://gitlab.com/groups/jx-gitlab-test/cluster"
    },
    "container_registry_image_prefix": "registry.gitlab.com/jx-gitlab-test/cluster/gitlab-import-test-1",
    "_links": {
        "self": "https://gitlab.com/api/v4/projects/31639063",
        "issues": "https://gitlab.com/api/v4/projects/31639063/issues",
        "merge_requests": "https://gitlab.com/api/v4/projects/31639063/merge_requests",
        "repo_branches": "https://gitlab.com/api/v4/projects/31639063/repository/branches",
        "labels": "https://gitlab.com/api/v4/projects/31639063/labels",
        "events": "https://gitlab.com/api/v4/projects/31639063/events",
        "members": "https://gitlab.com/api/v4/projects/31639063/members"
    },
    "packages_enabled": true,
    "empty_repo": false,
    "archived": false,
    "visibility": "private",
    "resolve_outdated_diff_discussions": false,
    "container_expiration_policy": {
        "cadence": "1d",
        "enabled": false,
        "keep_n": 10,
        "older_than": "90d",
        "name_regex": ".*",
        "name_regex_keep": null,
        "next_run_at": "2021-11-27T16:41:47.294Z"
    },
    "issues_enabled": true,
    "merge_requests_enabled": true,
    "wiki_enabled": true,
    "jobs_enabled": true,
    "snippets_enabled": true,
    "container_registry_enabled": true,
    "service_desk_enabled": true,
    "service_desk_address": "contact-project+jx-gitlab-test-cluster-gitlab-import-test-1-31639063-issue-@incoming.gitlab.com",
    "can_create_merge_request_in": true,
    "issues_access_level": "enabled",
    "repository_access_level": "enabled",
    "merge_requests_access_level": "enabled",
    "forking_access_level": "enabled",
    "wiki_access_level": "enabled",
    "builds_access_level": "enabled",
    "snippets_access_level": "enabled",
    "pages_access_level": "private",
    "operations_access_level": "enabled",
    "analytics_access_level": "enabled",
    "container_registry_access_level": "enabled",
    "emails_disabled": null,
    "shared_runners_enabled": true,
    "lfs_enabled": true,
    "creator_id": 3806623,
    "import_status": "none",
    "import_error": null,
    "open_issues_count": 0,
    "ci_default_git_depth": 50,
    "ci_forward_deployment_enabled": true,
    "ci_job_token_scope_enabled": false,
    "public_jobs": true,
    "build_git_strategy": "fetch",
    "build_timeout": 3600,
    "auto_cancel_pending_pipelines": "enabled",
    "build_coverage_regex": null,
    "ci_config_path": "",
    "shared_with_groups": [],
    "only_allow_merge_if_pipeline_succeeds": false,
    "allow_merge_on_skipped_pipeline": null,
    "restrict_user_defined_variables": false,
    "request_access_enabled": true,
    "only_allow_merge_if_all_discussions_are_resolved": false,
    "remove_source_branch_after_merge": true,
    "printing_merge_request_link_enabled": true,
    "merge_method": "merge",
    "squash_option": "default_off",
    "suggestion_commit_message": null,
    "merge_commit_template": null,
    "auto_devops_enabled": false,
    "auto_devops_deploy_strategy": "continuous",
    "autoclose_referenced_issues": true,
    "keep_latest_artifact": true,
    "external_authorization_classification_label": "",
    "requirements_enabled": false,
    "security_and_compliance_enabled": true,
    "compliance_frameworks": [],
    "permissions": {
        "project_access": null,
        "group_access": {
            "access_level": 50,
            "notification_level": 3
        }
    }
}
