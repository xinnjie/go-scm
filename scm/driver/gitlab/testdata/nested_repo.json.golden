{
    "ID": "31639063",
    "Namespace": "jx-gitlab-test",
    "Name": "cluster/gitlab-import-test-1",
    "FullName": "jx-gitlab-test/cluster/gitlab-import-test-1",
    "Perm": {
        "Pull": true,
        "Push": true,
        "Admin": true
    },
    "Branch": "main",
    "Private": true,
    "Clone": "https://gitlab.com/jx-gitlab-test/cluster/gitlab-import-test-1.git",
    "CloneSSH": "git@gitlab.com:jx-gitlab-test/cluster/gitlab-import-test-1.git",
    "Link": "https://gitlab.com/jx-gitlab-test/cluster/gitlab-import-test-1",
    "Created": "0001-01-01T00:00:00Z",
    "Updated": "0001-01-01T00:00:00Z"
}
